bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/io/ChannelPoolTest.java

```diff
@@ -31,35 +33,62 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
-import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 
+import com.google.cloud.bigtable.grpc.io.ChannelPool.WrappedChannel;
+
 
 @RunWith(JUnit4.class)
 @SuppressWarnings({"rawtypes", "unchecked"})
 public class ChannelPoolTest {
 
-  @Mock
-  private Channel channel;
-  @Mock
-  private MethodDescriptor descriptor;
-  @Mock
-  private ClientCall callStub;
-  @Mock
-  private ClientCall.Listener responseListenerStub;
-  @Mock
-  private HeaderInterceptor interceptor;
-
   @Test
   public void testInterceptorIsCalled() throws Exception {
-    MockitoAnnotations.initMocks(this);
-    when(channel.newCall(any(MethodDescriptor.class), any(CallOptions.class))).thenReturn(
+    Channel channel = mock(Channel.class);
+    MethodDescriptor descriptor = mock(MethodDescriptor.class);
+    ClientCall callStub = mock(ClientCall.class);
+    HeaderInterceptor interceptor = mock(HeaderInterceptor.class);
+    when(channel.newCall(same(descriptor), same(CallOptions.DEFAULT))).thenReturn(
       callStub);
     ChannelPool pool =
         new ChannelPool(new Channel[] { channel }, Collections.singletonList(interceptor));
     ClientCall call = pool.newCall(descriptor, CallOptions.DEFAULT);
     Metadata headers = new Metadata();
     call.start(null, headers);
-    verify(interceptor, times(1)).updateHeaders(eq(headers));
+    verify(interceptor, times(1)).updateHeaders(same(headers));
+  }
+
+  @Test
+  public void testChannelsAreRoundRobinned() {
+    Channel channel1 = mock(Channel.class);
+    Channel channel2 = mock(Channel.class);
+    MethodDescriptor descriptor = mock(MethodDescriptor.class);
+    MockitoAnnotations.initMocks(this);
+    ChannelPool pool = new ChannelPool(new Channel[] { channel1, channel2 }, null);
+    pool.newCall(descriptor, CallOptions.DEFAULT);
+    verify(channel1, times(1)).newCall(same(descriptor), same(CallOptions.DEFAULT));
+    verify(channel2, times(0)).newCall(same(descriptor), same(CallOptions.DEFAULT));
+    pool.newCall(descriptor, CallOptions.DEFAULT);
+    verify(channel1, times(1)).newCall(same(descriptor), same(CallOptions.DEFAULT));
+    verify(channel2, times(1)).newCall(same(descriptor), same(CallOptions.DEFAULT));
+  }
+  
+  @Test
+  public void testReserveReturnl() {
+    ChannelPool pool = new ChannelPool(new Channel[] { null, null }, null);
+    assertEquals(2, pool.size());
+    WrappedChannel reserved = pool.reserveChannel();
+    assertEquals(1, pool.size());
+    reserved.returnToPool();
+    assertEquals(2, pool.size());
+  }
+
+  @Test
+  public void testReserveNeverExhaustsPool() {
+    ChannelPool pool = new ChannelPool(new Channel[] { null, null }, null);
+    for (int i = 0; i < 10; i++) {
```

Comment:
nit:
pool.reserveChannel();
for (int i = 0; i < 10; i++) {
  assertEquals(1, pool.size());
}